File:  clojure/bob/1/bob.clj
(ns bob)

(defn- yell? [s]
  (and (re-find #"[a-zA-Z]+" s)
       (= (.toUpperCase s) s)))

(defn- question? [s]
  (.endsWith s "?"))

(defn- sentence? [s]
  (re-find #"\S+" s))

(defn response-for [s]
  (cond
    (yell? s) "Whoa, chill out!"
    (question? s) "Sure."
    (sentence? s) "Whatever."
    :else "Fine. Be that way!"))

File:  go/bob/1/bob.go
package bob

import (
	"strings"
	"unicode"
)

func Hey(remark string) string {
	remark = strings.TrimSpace(remark)

	if len(remark) == 0 {
		return "Fine. Be that way!"
	}

	if remark == strings.ToUpper(remark) && strings.ContainsFunc(remark, unicode.IsLetter) {
		if strings.HasSuffix(remark, "?") {
			return "Calm down, I know what I'm doing!"
		}
		return "Whoa, chill out!"
	}

	if strings.HasSuffix(remark, "?") {
		return "Sure."
	}

	return "Whatever."
}
